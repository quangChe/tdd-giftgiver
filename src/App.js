import React from 'react';
import './App.scss';
import { Button } from 'react-bootstrap';
import Gift from './components/Gift';


class App extends React.Component {
  state = {
    gifts: [],
  }

  addGift = (e) => {
    const { gifts }= this.state;
    const ids = gifts.map(gift => gift.id);
    const newId = ids.length ? Math.max(...ids) : 0;
    return this.setState({gifts: gifts.concat({id: newId + 1})});
  }

  removeGift = (id) => {
    const gifts = this.state.gifts.filter(gift => gift.id !== id);
    this.setState({ gifts })
  }

  render() { 
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className='gift-list'>
          {
            this.state.gifts.map(gift => (
              <Gift 
                key={gift.id}
                gift={gift}
                removeGift={this.removeGift} />
            ))
          }
        </div>
        <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
      </div>
    );
  }
}

export default App;
