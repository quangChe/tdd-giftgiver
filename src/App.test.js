import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const app = shallow(<App />);

it('renders without crashing', () => {
  expect(app).toMatchSnapshot();
});
