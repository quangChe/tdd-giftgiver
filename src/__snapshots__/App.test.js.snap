// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <Button
    active={false}
    className="btn-add"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Add Gift
  </Button>
</div>
`;
